hnn_core/network.py

```diff
@@ -219,16 +211,43 @@ def __repr__(self):
               % (self.n_of_type['L2_basket'], self.n_of_type['L5_basket']))
         return '<%s | %s>' % (class_name, s)
 
+    def _assign_external_feeds_coords(self, n_common_feeds, p_unique_keys):
```

Comment:
I would say, just move the contents of this into the method below to avoid more spaghetti than what already exists ... make a function when you use something more than once.

Reply:
We probably want pizza code: https://en.wikipedia.org/wiki/Spaghetti_code#Pizza_code 😄

Reply:
Right, will do. I'm still investigating what makes sense. I'm pretty stoked about getting this far though! I'll keep on digging, thankful for tests :)